### PR TITLE
Splitting of the projections gallery into projection types and complete the examples for azimuthal projections

### PR DESCRIPTION
**Pull request by xibalba01**
This pull request adds sub sections in the projection examples gallery for the five main projection groups that GMT supports and completes the azimuthal projection examples.

Related to #292 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: examples/projections/azim/azim_general_perspective.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+"""
+General Perspective
+===================
+
+The general perspective projection imitates the view of the earth from a finite
```
Comment (leouieda) [inline examples/projections/azim/azim_general_perspective.py]:
```suggestion
The general perspective projection imitates the view of the Earth from a finite
```

File: examples/projections/azim/azim_general_stereographic.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+"""
+General Stereographic
+=====================
+
+This map projection is a conformal, azimuthal projection. It is mainly used with
+a projection center in one of the poles. Then meridians appear as straight lines
+and cross latitudes at a right angle. Unlike the azimuthal equidistant projection
```
Comment (leouieda) [inline examples/projections/azim/azim_general_stereographic.py]:
```suggestion
and cross latitudes at a right angle. Unlike the azimuthal equidistant projection,
```

File: examples/projections/azim/azim_general_stereographic.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+the distances in this projection are not displayed in correct proportions.
+It is often used as a hemisphere map like the Lambert Azimuthal Equal Area
+projection.
+
+``slon0/lat0[/horizon]/scale`` or ``Slon0/lat0[/horizon]/width``
+
+``lon0/lat0`` specifies the projection center, the optional ``horizon`` parameter
+specifies the max distance from projection center (in degrees, < 180, default 90).
+
+This projection can be displayed
```
Comment (leouieda) [inline examples/projections/azim/azim_general_stereographic.py]:
```suggestion
This projection can be displayed:
```

File: examples/projections/azim/azim_general_stereographic.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+* with map boundaries coinciding with longitude and latitude:
```
Comment (leouieda) [inline examples/projections/azim/azim_general_stereographic.py]:
```suggestion
* With map boundaries coinciding with longitude and latitude:
```

File: examples/projections/azim/azim_general_stereographic.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  ``region`` specified via ``xmin/xmax/ymin/ymax``
+
```
Comment (leouieda) [inline examples/projections/azim/azim_general_stereographic.py]:
```suggestion
```

File: examples/projections/azim/azim_general_stereographic.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+* as a map with rectangular boundaries: ``region`` specified as lower left and
```
Comment (leouieda) [inline examples/projections/azim/azim_general_stereographic.py]:
```suggestion
* As a map with rectangular boundaries: ``region`` specified as lower left and
```

File: examples/projections/azim/azim_orthographic.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
+"""
+Orthographic
+============
+
+This projection is a perspective projection like the general perspective,
```
Comment (leouieda) [inline examples/projections/azim/azim_orthographic.py]:
```suggestion
This is a perspective projection like the general perspective,
```

File: examples/projections/azim/azim_orthographic.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
+"""
+Orthographic
+============
+
+This projection is a perspective projection like the general perspective,
+but with the difference that the point of perspective lies in infinite distance.
+It is therefore often used to give the appearance of a globe viewed from outer
+space, were one hemisphere can be seen as a whole. It is nor conformal, nor
```
Comment (leouieda) [inline examples/projections/azim/azim_orthographic.py]:
```suggestion
space, were one hemisphere can be seen as a whole. It is neither conformal nor
```

**Comment by leouieda** [comment]
Thanks @xibalba01! This was great. Looking forward to new contributions in the future :+1:

**Comment by xibalba01** [comment]
Glad i could help a bit @leouieda! Looking forward, too!

**Pull request merged**